custom_components/opnsense/coordinator.py

```
@@ -51,58 +52,44 @@
-    @_log_timing
-    async def _get_notices(self):
-        return await self._client.get_notices()
+    async def _get_states(self, categories: list) -> Mapping[str, Any]:
+        state: Mapping[str, Any] = {}
+        tasks: list = []
+        for cat in categories:
+            method = getattr(self._client, cat.get("function", ""), None)
+            if method:
+                tasks.append(method())
+            else:
+                _LOGGER.error(f"Method {cat.get('function','')} not found.")
```

Ruff

Logging statement uses f-string

(G004)